uiowa/uiowa
# Prospector webform handler

Resolves #8138 

# How to test
```
ddev blt ds --site education.uiowa.edu && ddev drush @education.local config-split:activate advanced_webform && ddev drush @education.local uli
```
- Test that error messages are working by adding the Prospector handler to a webform. It should display messages about auth credentials, site UUID. If you want to test the error message related to endpoint URL, you'll need to figure out the best way to temporarily unset the config value for it, which should be set when the site syncs.
- Add the following to the `docroot/sites/education.uiowa.edu/settings/local.settings.php` and replacing the `username` and `password` with the ones stored in esaccess.
```php
$config['sitenow_advanced_webform.settings']['prospector']['auth']['user'] = 'username';
$config['sitenow_advanced_webform.settings']['prospector']['auth']['pass'] = 'password';
```
- Visit https://education.uiowa.ddev.site/admin/config/sitenow/advanced-webform. There is a pinned email in the web team inbox that contains the `siteInteractionUuid` for education.uiowa.edu. This should be entered into the settings form.
- Add the Prospector handler to a webform that ideally already has `first_name`, `last_name`, `email`, and `phone`. Note that the error messages from before are no longer present.
- Submit the form and observe that there is a message indicating that the data was successfully transferred to the application.
- To test error logging, enable dblog locally and comment out line 141 in `Drupal\sitenow_advanced_webform\Plugin\WebformHandler\ProspectorRemotePostWebformHandler`: `$options['json'] = $data;` and then resubmit the form. Visit the logs and see the error printed there.
